omod/src/main/java/org/openmrs/module/appointments/web/service/impl/AllAppointmentRecurringPatternUpdateService.java

```diff
@@ -21,16 +23,20 @@
     @Autowired
     private AppointmentMapper appointmentMapper;
 
+    @Autowired
+    private RecurringPatternMapper recurringPatternMapper;
+
     @Autowired
     RecurringAppointmentsService recurringAppointmentsService;
 
     public AppointmentRecurringPattern getUpdatedRecurringPattern(RecurringAppointmentRequest recurringAppointmentRequest) {
         String clientTimeZone = recurringAppointmentRequest.getTimeZone();
         String serverTimeZone = Calendar.getInstance().getTimeZone().getID();
         Appointment appointment = getAppointment(recurringAppointmentRequest);
+        AppointmentRecurringPattern recurringPattern = recurringPatternMapper.cloneAppointmentRecurringPattern(appointment.getAppointmentRecurringPattern());
         appointmentMapper.mapAppointmentRequestToAppointment(recurringAppointmentRequest.getAppointmentRequest(), appointment);
         List<Appointment> newSetOfAppointments = recurringAppointmentsService
-                .getUpdatedSetOfAppointments(appointment.getAppointmentRecurringPattern(), recurringAppointmentRequest);
+                .getUpdatedSetOfAppointments(recurringPattern, recurringAppointmentRequest);
```

Comment:
Does it always consider cloned `recurringPattern` object? Even in create/update flows?